### PR TITLE
[FE][BE] schedule bugfix

### PR DESCRIPTION
**Pull request by johnp418**
[WIP] Refactoring both front and backend code so that we only send related data when updating our entities.
- Fixed a bug where updating schedule was resetting start and last processed date.
- Daypicker in schedule form now disables past dates

### CODE REVIEW

File: backend/src/test/java/com/ez/ezbackend/budget/request/ScheduleRequestTest.java [hunk 1]
```diff
@@ -121,25 +130,29 @@ private void testConvertToScheduleWith(User user, Category category, Long updati
...
-    } else if (category != null && updatingScheduleId == null) {
-      schedule = ScheduleRequest.convertToSchedule(scheduleRequest, user, category);
-    } else if (category != null && updatingScheduleId != null) {
-      schedule = ScheduleRequest.convertToSchedule(scheduleRequest, user, category, updatingScheduleId);
-    }
+
+    Schedule previousSchedule = Schedule.builder()
+        .id(updatingScheduleId)
+        .startDate(LocalDate.of(2018, 1, 1))
+        .nextRecurringDate(LocalDate.of(2018,1,15))
```
Comment (mwkim0919) [inline backend/src/test/java/com/ez/ezbackend/budget/request/ScheduleRequestTest.java]: nitpicking, spaces after comma in `2018,1,15`

File: backend/src/main/java/com/ez/ezbackend/budget/request/ScheduleRequest.java [hunk 2]
```diff
@@ -30,21 +30,24 @@
...
     return convertToSchedule(scheduleRequest, user, category, null);
   }
 
-  public static Schedule convertToSchedule(ScheduleRequest scheduleRequest, User user, Category category, Long updatingScheduleId) {
-    validateSchedule(scheduleRequest);
+  public static Schedule convertToSchedule(ScheduleRequest scheduleRequest, User user, Category category, Schedule previousSchedule) {
+    validateSchedule(scheduleRequest, category, previousSchedule);
+    if (previousSchedule == null) {
+      return convertToScheduleForCreate(scheduleRequest, user, category);
+    } else {
```
Comment (mwkim0919) [inline backend/src/main/java/com/ez/ezbackend/budget/request/ScheduleRequest.java]: don't really need else statement.

File: backend/src/test/java/com/ez/ezbackend/budget/request/ScheduleRequestTest.java (hunk 1 above)
Comment (johnp418) [reply backend/src/test/java/com/ez/ezbackend/budget/request/ScheduleRequestTest.java]: Fixed.

File: backend/src/main/java/com/ez/ezbackend/budget/request/ScheduleRequest.java (hunk 2 above)
Comment (johnp418) [reply backend/src/main/java/com/ez/ezbackend/budget/request/ScheduleRequest.java]: Fixed.

**Pull request merged**